### ISSUE TITLE
Function get_relevant_docs() returning AttributeError

### ISSUE THREAD

**Issue by gabriellm1** [langchain-ai/langchain#12076]
### System Info

MacOS Ventura 13.6
Python 3.10.13
langchain 0.0.306

### Who can help?

@hwchase17
@agola11 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```
  llm = AzureOpenAI(temperature=0, model="gpt-35-turbo")
  compressor = LLMChainExtractor.from_llm(llm)
  base_retriever = vectorstores.as_retriever()
  compression_retriever = ContextualCompressionRetriever(base_compressor=compressor, base_retriever=base_retriever)
  compression_retriever.get_relevant_documents("Owner")
```

### Expected behavior

Expected to return the docs compressed from the vectorstore, but I'm getting the `AttributeError: 'str' object has no attribute 'get'`

**Comment by gabriellm1** [comment]
I do not think that is the problem since I instantiated a retriever using the method `as_retriever()` in the FAISS vectorstore

**Comment by zhuzihan728** [comment]
This can happen when you receive an HTTP error response, e.g., 404, from OpenAI embedding API.
